src/context/WalletProvider/NewWalletViews/NewWalletViewsSwitch.tsx

```diff
@@ -0,0 +1,276 @@
+import { ArrowBackIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Flex,
+  IconButton,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+  useToast,
+} from '@chakra-ui/react'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { isMobile } from 'react-device-detect'
+import { useTranslate } from 'react-polyglot'
+import { Route, Switch, useHistory } from 'react-router-dom'
+import { Text } from 'components/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { SnapInstall } from '../MetaMask/components/SnapInstall'
+import { SnapUpdate } from '../MetaMask/components/SnapUpdate'
+import { NativeWalletRoutes } from '../types'
+import { InstalledWalletsSection } from './sections/InstalledWalletsSection'
+import { MipdBody } from './wallets/mipd/MipdBody'
+
+const sectionsWidth = { base: 'full', md: '300px' }
+const containerWidth = {
+  base: 'full',
+  md: '900px',
+}
+
+const arrowBackIcon = <ArrowBackIcon />
+
+const INITIAL_WALLET_MODAL_ROUTE = '/'
+
+type RightPanelContentProps = {
+  isLoading: boolean
+  setIsLoading: (loading: boolean) => void
+  error: string | null
+  setError: (error: string | null) => void
+}
+
+const RightPanelContent = ({
+  isLoading,
+  setIsLoading,
+  error,
+  setError,
+}: RightPanelContentProps) => {
+  const {
+    state: { modalType, isMipdProvider },
+  } = useWallet()
+
+  if (isMipdProvider && modalType) {
+    return (
+      <Switch>
+        <Route exact path='/metamask/connect'>
+          <MipdBody
+            rdns={modalType}
+            isLoading={isLoading}
+            error={error}
+            setIsLoading={setIsLoading}
+            setError={setError}
+          />
+        </Route>
+        <Route path='/metamask/snap/install'>
+          <Flex direction='column' alignItems='center' justifyContent='center' height='full'>
+            <SnapInstall />
+          </Flex>
+        </Route>
+        <Route path='/metamask/snap/update'>
+          <Flex direction='column' alignItems='center' justifyContent='center' height='full'>
+            <SnapUpdate />
+          </Flex>
+        </Route>
+        <Route path='/'>
+          <MipdBody
+            rdns={modalType}
+            isLoading={isLoading}
+            error={error}
+            setIsLoading={setIsLoading}
+            setError={setError}
+          />
+        </Route>
+      </Switch>
+    )
+  }
+
+  return null
```

Comment:
No default state for the time being, just an empty blank state which is ugly but temporary - @shapeshift/product we may want to add something like this
<img width="791" alt="image" src="https://github.com/user-attachments/assets/401f953a-a08d-4684-815d-5b63458ea67b" />

Reply:
@gomesalexandre The default state is in the figma, it tells them about the native wallet and allows them to create or import
https://www.figma.com/design/om2yDg0BjLNNSkJz82O30N/Wallet-Modal?node-id=2818-282&t=JAqFe0kHamer4ajp-0